%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Beamer Presentation
% LaTeX Template
% Version 1.0 (10/11/12)
%
% This template has been downloaded from:
% http://www.LaTeXTemplates.com
%
% License:
% CC BY-NC-SA 3.0 (http://creativecommons.org/licenses/by-nc-sa/3.0/)
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%----------------------------------------------------------------------------------------
%	PACKAGES AND THEMES
%----------------------------------------------------------------------------------------

\documentclass[14pt,handout]{beamer}
%%\documentclass[14pt]{beamer}

\mode<presentation> {

% The Beamer class slide themes
\usetheme{Madrid} %i was using this one

% Beamer class color themes

%\usecolortheme{albatross}

%\setbeamertemplate{footline} % To remove the footer line in all slides uncomment this line
%\setbeamertemplate{footline}[page number] % To replace the footer line in all slides with a simple slide count uncomment this line

%\setbeamertemplate{navigation symbols}{} % To remove the navigation symbols from the bottom of all slides uncomment this line
}

\usepackage{graphicx} % Allows including images
\usepackage{booktabs} % Allows the use of \toprule, \midrule and \bottomrule in tables
\usepackage{hyperref}
\usepackage{helvet}
\usepackage[T1]{fontenc}
\usepackage{textcomp}

%----------------------------------------------------------------------------------------
%	TITLE PAGE
%----------------------------------------------------------------------------------------

\title[RNAseq Practical pt2/3]{RNAseq Analysis: A Practical Walkthrough (part 2 and a half/3)} % The short title appears at the bottom of every slide, the full title is only on the title page

\author{C. Ryan Campbell} % Your name
\institute[Duke] % Your institution as it will appear on the bottom of every slide, may be shorthand to save space
{
Duke University \\ % Your institution for the title page
\medskip
\textit{c.ryan.campbell@duke.edu} % Your email address
}
\date{31 Oct 2017} % Date, can be changed to a custom date

\begin{document}

\begin{frame}
\titlepage % Print the title page as the first slide
\end{frame}

\begin{frame}
\frametitle{Overview} % Table of contents slide, comment this block out to remove it
\tableofcontents % Throughout your presentation, if you choose to use \section{} and \subsection{} commands, these will automatically be printed on this slide as an overview of your presentation
\end{frame}

%----------------------------------------------------------------------------------------
%	PRESENTATION SLIDES
%----------------------------------------------------------------------------------------

%------------------------------------------------
\begin{frame}
\frametitle{Today's Goals}
\begin{itemize}
	\item<+-> Logon to interactive SLURM node
	\item<+-> Run tophat2
	\item<+-> Run htseq-count
\end{itemize}
\end{frame}

%------------------------------------------------
\section{Workflow}
%------------------------------------------------

%------------------------------------------------
\begin{frame}
\frametitle{Workflow}
\begin{enumerate}
	\large
	\item<+-> Fill in the missing blanks on my diagram
	\begin{itemize}
		\item<+-> Fill in either a file status or a software/process
		\item<+-> Should take about 1 minute
	\end{itemize}
\end{enumerate}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{slogin OR sbatch script}
\begin{itemize}
	\large
	\item<+-> What is the difference?
	\item<+-> Make sure you're making a conscious choice between the two
	\item<+-> Today we'll be working on slogin with \underline{SMALL} files
	\item<+-> (why does this matter?)
	\item<+-> When you analyze your files, make sure to use an sbatch script
\end{itemize}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{SLURM Interactive Node}
\begin{itemize}
	\item Later we'll be troubleshooting \texttt{htseq-count}
	\item For that you should use an ``interactive node''
	\item This runs like a sbatch job, but it appears as a terminal that you can interact with
	\footnotesize
	\ttfamily
	\begin{block}{}
	\item[] srun --mem-per-cpu=4000MB --pty bash -i
	\end{block}
	\sffamily
	\item You've just requested a 16GB (powerful laptop) size node on SLURM
\end{itemize}
\end{frame}

%------------------------------------------------
\section{Review Steps}
%------------------------------------------------

%------------------------------------------------
\begin{frame}
\frametitle{fastqc output}
\begin{columns}
	\begin{column}{0.35\textwidth}
		\begin{itemize}
			\item Run Quality
			\item Base Content
			\item Run Length
		\end{itemize}
		\end{column}
	\begin{column}{0.65\textwidth}
		\begin{center}
     		\includegraphics[width=1\textwidth]{/Users/ryan/Documents/git_repos/duke-bio490s/slides/images_20171012_run_qual.png}
     	\end{center}
	\end{column}
\end{columns}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{Trimming Data}
\begin{itemize}
	\item<+-> Remove data that is low quality
	\begin{itemize}
		\item<+-> You have \underline{TONS} of data, taking out 5\% is OK
	\end{itemize}
	\item<+-> We will set a couple of parameters:
	\begin{itemize}
		\item<+-> Minimum quality at the end of the read
		\item<+-> Average quality along a sliding window
		\item<+-> Overall read quality
	\end{itemize}
	\item<+-> If the read doesn't meet some, or all, of these the whole read is tossed
	\item<+-> We'll be using \href{http://www.usadellab.org/cms/?page=trimmomatic}{trimmomatic}
\end{itemize}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{trimmomatic example}
\begin{itemize}
	\item To run trimmomatic:
	\ttfamily
	\footnotesize
	\begin{block}{}
	\item[] EXAMPLE:
	\item[] java -jar /work/cc216/490S/software/Trimmomatic-0.36/trimmomatic-0.36.jar <PE or SE> -phred33 -trimlog <output log> <Read 1.fq> <Read 2.fq> <Read 1 output> <Read 1 output unpaired> <Read 2 output> <Read 2 output unpaired> LEADING:3 TRAILING:3 SLIDINGWINDOW:5:20 MINLEN:50
	\end{block}
	\sffamily
	\item Remove (leading/trailing) low quality or N bases (below quality 3)
	\item Scan the read with a 5-base wide sliding window, cutting when the average quality per base drops below 20
	\item Drop reads which are less than 50 bases long after these steps
\end{itemize}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{trimmomatic example}
\begin{itemize}
	\item Output:
	\ttfamily
	\footnotesize
	\begin{block}{}
	\item[] Input Read Pairs: 1000000 Both Surviving: 955447 (95.54\%) Forward Only Surviving: 29029 (2.90\%) Reverse Only Surviving: 9577 (0.96\%) Dropped: 5947 (0.59\%)
	\item[] >head RNAseq.log
	\item[] SRR848963.63 ILLUMINA:322:D0UFKACXX:3:1101:11445:2184 length=101 101 0 101 0
	\item[] SRR848963.64 ILLUMINA:322:D0UFKACXX:3:1101:11909:2032 length=101 97 1 98 3
	\item[] SRR848963.64 ILLUMINA:322:D0UFKACXX:3:1101:11909:2032 length=101 98 0 98 3
	\end{block}
\end{itemize}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{Downloading genomes}
\begin{columns}
	\begin{column}{0.3\textwidth}
		\begin{itemize}
			\footnotesize
			\item<+-> Model species have a page like this
			\item<+-> Download from the ``FASTA format for genome'' and ``annotation in GFF'' links
		\end{itemize}
		\end{column}
	\begin{column}{0.7\textwidth}
		\begin{center}
     		\includegraphics[width=1\textwidth]{/Users/ryan/Documents/git_repos/duke-bio490s/slides/images_20171012_fly_genome.png}
     	\end{center}
	\end{column}
\end{columns}
\end{frame}

%------------------------------------------------
\subsection{Indexing}
%------------------------------------------------

%------------------------------------------------
\begin{frame}
\frametitle{Indexing}
\begin{itemize}
	\large
	\item<+-> Most aligners require their genome to be indexed
	\item<+-> What do you think this means?
	\item<+-> You'll need to index using bowtie2 (the aligner tophat2 uses)
\end{itemize}
\end{frame}

%------------------------------------------------
\section{Alignment}
%------------------------------------------------

%------------------------------------------------
\subsection{tophat}
%------------------------------------------------

%------------------------------------------------
\begin{frame}
\frametitle{tophat}
\begin{itemize}
	\item<+-> tophat2 - alignment software
	\item<+-> In: Sequence data
	\item<+-> Out: .bam file - where that data aligns to/fits in the genome
\end{itemize}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{tophat}
\begin{itemize}
	\item tophat2 is the command/software that aligns the reads to the genome
	\item This (also) is going to be a computationally intensive process
	\item So write a submission script to do it:
	\footnotesize
	\ttfamily
	\begin{block}{}
	\item[] export PATH=/opt/apps/tophat-bowtie/:\$PATH
	\item[]
	\item[] tophat2
	\end{block}
	\sffamily
\end{itemize}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{tophat}
\begin{itemize}
	\item tophat2 example (all one line):
	\footnotesize
	\ttfamily
	\begin{block}{}
	\item[]  tophat2 -p 4 -o s01 -G /work/cc216/490S/cc216/RNAseq\_pt2/hsap\_annotations.gff /work/cc216/490S/cc216/RNAseqpt2/hsap /work/cc216/490S/cc216/RNAseq\_pt2/s01\_hsap\_norm\_R1.fastq /work/cc216/490S/cc216/RNAseq\_pt2/s01\_hsap\_norm\_R2.fastq
	\item[] 
	\end{block}
	\sffamily
	\item[] translated:
	\ttfamily
	\begin{block}{}
	\item[]  tophat2 -p <number of threads> -o <output dir> -G <gff file, annotations> <bowtie2 index> <R1 fastq> <R2 fastq>
	\item[] 
	\end{block}
	\sffamily
	\item Help can be found by running ``tophat2''
	\item Or in the tophat2 manual online
	\item \href{http://ccb.jhu.edu/software/tophat/manual.shtml}{http://ccb.jhu.edu/software/tophat/manual.shtml}
\end{itemize}
\end{frame}

%------------------------------------------------
\section{Counting}
%------------------------------------------------

%------------------------------------------------
\begin{frame}
\frametitle{Counting \& Analysis}
\begin{itemize}
	\item<+-> Now that we have bam files the next step is to count the reads
	\item<+-> And using those counts compare gene expression levels
	\item<+-> We'll be using HTseq
\end{itemize}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{HTSeq}
\begin{itemize}
	\item<+-> python-based program to count reads
	\item<+-> Input: 
	\item<+-> .bam file \underline{and} .gtf/.gff
	\item<+-> Output:
	\item<+-> A table of counts by gene
\end{itemize}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{htseq-count}
\begin{itemize}
	\item We'll be using htseq-count
	\item This will count the number of reads mapped to each gene
	\item That data will be taken into DESeq2
	\footnotesize
	\ttfamily
	\begin{block}{}
	\item[] /opt/apps/rhel7/Python-2.7.11/bin/htseq-count 
	\end{block}
	\sffamily
	\item[] (go ahead and put it in your path)
\end{itemize}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{htseq-count}
\begin{itemize}
	\item What does HTSeq do?
	\item What are its flags and options?
	\footnotesize
	\ttfamily
	\begin{block}{}
	\item[] htseq-count <options> <alignment bam> <gff file>  >  <count output>
	\end{block}
	\sffamily
	\item Probably important: -f, -s, -t
\end{itemize}
\end{frame}

%------------------------------------------------
\section{Tutorial Files}
%------------------------------------------------

%------------------------------------------------
\begin{frame}
\frametitle{Files to Use}
\begin{itemize}
	\item I've set up some example files to use for this tutorial
	\item They're human RNAseq files from a hypoxia experiment:
	\footnotesize
	\ttfamily
	\begin{block}{}
	\item[] ls -lthr /work/cc216/490S/cc216/RNAseq\_pt2
	\end{block}
	\sffamily
	\item What do you see? Which will you use?
\end{itemize}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{Files to Use}
\begin{itemize}
	\footnotesize
	\ttfamily
	\begin{block}{}
	\item[] ls -l /work/cc216/490S/cc216/RNAseq\_pt2
	\end{block}
	\begin{center}
		\includegraphics[width=.9\textwidth]{images_20171031_usefiles.png}
	\end{center}
\end{itemize}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{symbolic links}
\begin{itemize}
	\item Some of the files in this directory are symbolic links or ``sym-links''
	\item It looks like the file is there, but it actually just points to the true location
	\item Essentially a shortcut, you can call the link as if it were the file
	\footnotesize
	\ttfamily
	\begin{block}{}
	\item[] ln -s <file to link to> <link to create>
	\item[] 
	\item[] ln -s s01/accepted\_hits.bam ./s01\_accepted\_hits.bam
	\end{block}
	\sffamily
	\item[] (the -s stands for ``soft'' which allows links across directories and file systems)
\end{itemize}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{SLURM Interactive Node}
\begin{itemize}
	\item Now that we're about to troubleshoot \texttt{htseq-count} hopefully your SLURM node is open
	\footnotesize
	\ttfamily
	\begin{block}{}
	\item[] srun --mem-per-cpu=4000MB --pty bash -i
	\end{block}
	\sffamily
\end{itemize}
\end{frame}

%------------------------------------------------
\subsection{Commands}
%------------------------------------------------

%------------------------------------------------
\begin{frame}
\frametitle{htseq-count}
\begin{itemize}
	\item What does HTSeq do?
	\item What are its flags and options?
	\footnotesize
	\ttfamily
	\begin{block}{}
	\item[] htseq-count <options> <alignment bam> <gff file> > <count output>
	\end{block}
	\sffamily
	\item Probably important: -f, -s, -t
\end{itemize}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{tophat}
\begin{itemize}
	\item tophat2 example (all one line):
	\footnotesize
	\ttfamily
	\begin{block}{}
	\item[]  tophat2 -p 4 -o s01 -G /work/cc216/490S/cc216/RNAseq\_pt2/hsap\_annotations.gff /work/cc216/490S/cc216/RNAseqpt2/hsap /work/cc216/490S/cc216/RNAseq\_pt2/s01\_hsap\_norm\_R1.fastq /work/cc216/490S/cc216/RNAseq\_pt2/s01\_hsap\_norm\_R2.fastq
	\item[] 
	\end{block}
	\sffamily
	\item[] translated:
	\ttfamily
	\begin{block}{}
	\item[]  tophat2 -p <number of threads> -o <output dir> -G <gff file, annotations> <bowtie2 index> <R1 fastq> <R2 fastq>
	\item[] 
	\end{block}
	\sffamily
	\item Help can be found by running ``tophat2''
	\item Or in the tophat2 manual online
	\item \href{http://ccb.jhu.edu/software/tophat/manual.shtml}{http://ccb.jhu.edu/software/tophat/manual.shtml}
\end{itemize}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{Today's Goals}
\large
\begin{enumerate}
	\item<+-> Logon to interactive SLURM node
	\item<+-> Run tophat2
	\item<+-> Run htseq-count
\end{enumerate}
\end{frame}

%------------------------------------------------
\subsection{DESeq2}
%------------------------------------------------

%------------------------------------------------
\begin{frame}
\frametitle{DESeq2}
\begin{itemize}
	\item<+-> What does DESeq2 do?
	\item<+-> Compares the count matrices from many samples
	\item<+-> Where do you run it?
	\item<+-> In R on your laptop
\end{itemize}
\end{frame}

%------------------------------------------------
\begin{frame}
\frametitle{DESeq2}
\begin{itemize}
	\item<+-> R-based program to analyze expression
	\item<+-> Input: 
	\item<+-> A table of counts by gene
	\item<+-> Output: 
	\item<+-> Graphs and (hopefully) Answers!!!
\end{itemize}
\end{frame}


%------------------------------------------------
\begin{frame}
\frametitle{DESeq2 guides}
\begin{itemize}
	\item<+-> We'll get into DESeq2 next week
	\item<+-> If you want to get started here are some guides:
	\item<+-> \href{http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html}{Walkthrough}
	\item<+-> \href{https://www.bioconductor.org/packages/release/bioc/html/DESeq2.html}{Bioconductor Manual} 
	\item<+-> \href{https://www.bioconductor.org/help/workflows/rnaseqGene/}{Bioconductor Walkthrough}
\end{itemize}
\end{frame}


%------------------------------------------------
\begin{frame}
\Huge{\centerline{The End}}
\end{frame}

%----------------------------------------------------------------------------------------

\end{document} 